### PR TITLE
Add ethernet.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/net/ethernet.toit [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          --mac_mdc=mac_mdc
+          --mac_mdio=mac_mdio
+          --mac_spi_device=mac_spi_device
+          --mac_int=mac_int
+      try:
+        with_timeout ETHERNET_CONNECT_TIMEOUT_: ethernet.connect
+        with_timeout ETHERNET_DHCP_TIMEOUT_: ethernet.get_ip
+        // Success: Register the ethernet connection, tell anyone who is waiting
+        // for it that the connection is ready to be used (no exception), and
+        // go on to mark ourselves as a user of the WiFi network.
```
Comment (erikcorry) [inline lib/net/ethernet.toit:110]:
```suggestion
        // go on to mark ourselves as a user of the Ethernet network.
```

File: lib/net/modules/ethernet.toit [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
```
Comment (erikcorry) [inline lib/net/modules/ethernet.toit:4]:
```suggestion

// Driver for wired Ethernet.

```

**Review by erikcorry — APPROVED** [review]

File: lib/net/ethernet.toit [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+export PHY_CHIP_NONE PHY_CHIP_IP101 PHY_CHIP_LAN8720
+
+ETHERNET_CONNECT_TIMEOUT_  ::= Duration --s=10
+ETHERNET_DHCP_TIMEOUT_     ::= Duration --s=16
+
+ethernet_/ethernet.Ethernet? := null
+ethernet_connecting_/monitor.Latch? := null
+
+/**
+Connects the ethernet peripheral.
```
[the remark this answers is not in the document]
Comment (floitsch) [reply lib/net/ethernet.toit]: done.

File: lib/net/ethernet.toit [hunk 4]
```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
```
[the remark this answers is not in the document]
Comment (floitsch) [reply lib/net/ethernet.toit:4]: Added a toitdoc comment.

File: lib/net/ethernet.toit [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+          --phy_reset=phy_reset
+          --mac_chip=mac_chip
+          --mac_mdc=mac_mdc
+          --mac_mdio=mac_mdio
+          --mac_spi_device=mac_spi_device
+          --mac_int=mac_int
+      try:
+        with_timeout ETHERNET_CONNECT_TIMEOUT_: ethernet.connect
+        with_timeout ETHERNET_DHCP_TIMEOUT_: ethernet.get_ip
+        // Success: Register the ethernet connection, tell anyone who is waiting
```
[the remark this answers is not in the document]
Comment (floitsch) [reply lib/net/ethernet.toit]: done.

File: lib/net/ethernet.toit (hunk 1 above)
Comment (floitsch) [reply lib/net/ethernet.toit]: done.

File: lib/net/modules/ethernet.toit (hunk 2 above)
Comment (floitsch) [reply lib/net/modules/ethernet.toit:4]: added a toitdoc comment.

**Pull request merged**